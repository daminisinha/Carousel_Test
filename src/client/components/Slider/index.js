import React, { Component, Fragment } from 'react';
import {Visible} from 'react-grid-system'
class Slide extends Component {
    constructor(props) {
        super(props);
        this.state = {};
    }
    render() {
        let { image, width, height, index } = this.props;
        let alt = image.tags? image.tags.split(',').join(' and '):'image';
  
        return (
            <Fragment>
                <Visible md lg xl>
                <img width={width} alt={alt} onClick={() => this.props.onClickHandler(index)} height={height} src={image.webformatURL} />
                </Visible>
                <Visible xs sm>
                <img width={width} alt={alt} onClick={() => this.props.onClickHandler(index)} src={image.webformatURL} />
                </Visible>
            </Fragment>
        );
    }
}

export default Slide;