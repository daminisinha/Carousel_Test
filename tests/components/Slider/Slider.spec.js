import React from 'react';
import Slider from '../../../src/client/components/Slider'
import {shallow} from 'enzyme';

test('Slider renders',()=>{
    let image ={
        tags :0
    }
    const component = shallow(<Slider image ={image.tags}/>); 
    expect(component).toMatchSnapshot();
})
