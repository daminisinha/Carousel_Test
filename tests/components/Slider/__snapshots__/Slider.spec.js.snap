// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders 1`] = `
<Fragment>
  <Visible
    lg={true}
    md={true}
    sm={false}
    xl={true}
    xs={false}
  >
    <img
      alt="image"
      onClick={[Function]}
    />
  </Visible>
  <Visible
    lg={false}
    md={false}
    sm={true}
    xl={false}
    xs={true}
  >
    <img
      alt="image"
      onClick={[Function]}
    />
  </Visible>
</Fragment>
`;
